package/MDAnalysis/analysis/base.py

```diff
@@ -24,61 +24,116 @@
 
 logger = logging.getLogger(__name__)
 
 
 class AnalysisBase(object):
-    """Base class for defining multi frame analysis
-
-    The analysis base class is designed as a template for creating
-    multiframe analysis.
-
-    The class implements the following methods:
-
-    _setup_frames(trajectory, start=None, stop=None, step=None)
-      Pass a Reader object and define the desired iteration pattern
-      through the trajectory
-
-    run
-      The user facing run method.  Calls the analysis methods
-      defined below
-
-    Your analysis can implement the following methods, which are
-    called from run:
-
-    _prepare
-      Called before iteration on the trajectory has begun.
-      Data structures can be set up at this time, however most
-      error checking should be done in the __init__
-
-    _single_frame
-      Called after the trajectory is moved onto each new frame.
-
-    _conclude
-      Called once iteration on the trajectory is finished.
-      Apply normalisation and averaging to results here.
+    """Base class for defining multi frame analysis, it is designed as a
+    template for creating multiframe analysis. This class will automatically
+    take care of setting up the trajectory reader for iterating and offers to
+    show a progress meter.
+
+    To define a new Analysis, `AnalysisBase` needs to be subclassed
+    `_single_frame` must be defined. It is also possible to define
+    `_prepare` and `_conclude` for pre and post processing. See the example
+    below.
+
+    .. code-block:: python
+       class NewAnalysis(AnalysisBase):
+           def __init__(self, atomgroup, parameter, **kwargs):
+               super(NewAnalysis, self).__init__(atomgroup.universe.trajectory,
+                                                 **kwargs)
+               self._parameter = parameter
+               self._ag = atomgroup
+
+           def _prepare(self):
+               # OPTIONAL
+               # Called before iteration on the trajectory has begun.
+               # Data structures can be set up at this time
+               self.result = []
+
+           def _single_frame(self):
+               # REQUIRED
+               # Called after the trajectory is moved onto each new frame.
+               # store result of `some_function` for a single frame
+               self.result.append(some_function(self._ag, self._parameter))
+
+           def _conclude(self):
+               # OPTIONAL
+               # Called once iteration on the trajectory is finished.
+               # Apply normalisation and averaging to results here.
+               self.result = np.asarray(self.result) / np.sum(self.result)
+
+    Afterwards the new analysis can be run like this.
+
+    .. code-block:: python
+       na = NewAnalysis(u.select_atoms('name CA'), 35).run()
+       print(na.result)
 
     """
+    def __init__(self, trajectory, start=None,
+                 stop=None, step=None, quiet=True):
```

Comment:
Yeah we can do that. LIkely something we should do before 1.0.0. Can you open a bug for this. The deprecation is easy to do just tedious to catch all the instances where we use quiet.